src/fluree/db/query/json_ld/response.cljc

```diff
@@ -80,68 +80,71 @@
+               _     (log/debug "flakes->res spec:" spec)
+               p-iri (:as spec)
+               v     (cond
+                       (nil? spec)
+                       nil
+
+                       ;; flake's .-o value is an IRI string, JSON-LD compact it before returning
+                       (iri? p)
+                       (-> p-flakes first flake/o compact-fn)
+
+                       ;; flake's .-o value is a rdf:type, resolve subject id to IRI then JSON-LD compact it
+                       (rdf-type? p)
+                       (loop [[type-id & rest-types] (map flake/o p-flakes)
+                              acc []]
+                         (if type-id
+                           (recur rest-types
+                                  (conj acc (or (get @cache type-id)
+                                                (<? (cache-sid->iri db cache compact-fn type-id)))))
+                           acc))
+
+                       :else                               ;; display all values
+                       (loop [[f & r] (if list?
+                                        (sort-by #(:i (flake/m %)) p-flakes)
+                                        p-flakes)
+                              acc []]
+                         (if f
+                           (let [res (if (= const/$xsd:anyURI (flake/dt f))
+                                       (cond
+                                         ;; have a specified sub-selection (graph crawl)
+                                         (:spec spec)
+                                         (<? (crawl-ref-item db context compact-fn (flake/o f) (:spec spec) cache fuel-vol max-fuel (inc depth-i)))
+
+                                         ;; requested graph crawl depth has not yet been reached
+                                         (< depth-i depth)
+                                         (<? (crawl-ref-item db context compact-fn (flake/o f) select-spec cache fuel-vol max-fuel (inc depth-i)))
+
+                                         ;; no sub-selection, just return {@id <iri>} for each ref iri
+                                         :else
+                                         ;; TODO - we generate id-key here every time, this should be done in the :spec once beforehand and used from there
+                                         (let [id-key (:as (wildcard-spec db cache compact-fn const/$xsd:anyURI))
+                                               c-iri  (<? (dbproto/-iri db (flake/o f) compact-fn))]
+                                           (log/debug "id-key:" id-key "c-iri:" c-iri)
+                                           {id-key c-iri}))
+                                       (flake/o f))]
+                             (recur r (conj acc res)))
+                           (if (and (= 1 (count acc))
+                                    (not (#{"list" "set" :list :set} (-> context (get p-iri) (get :container)))))
```

Comment:
Oh, we likely don't need all of those anymore. Good catch!

Reply:
That's because our `json-ld/expand` function keywordizes `"@container"` and `"@list"` and removes the "@" prefix from `"@set"`. This code is just meant to work with that output.